From: Michal Kubecek <mkubecek@suse.cz>
Date: Tue, 28 May 2013 11:52:53 +0200
Subject: netfilter: ip6t_LOG: fix logging of packet mark
Patch-mainline: Queued in subsystem maintainer repository
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/davem/net.git
Git-commit: d660164d79b67f879db35a7d61e47d3b99bc714e
References: bnc#821930

In dump_packet(), the "recurse" parameter is zero only if dumping
contents of a packet embedded into an ICMPv6 error message. Thus
we want to log packet mark if recurse is non-zero, not when it is
zero.

Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 net/ipv6/netfilter/ip6t_LOG.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ipv6/netfilter/ip6t_LOG.c b/net/ipv6/netfilter/ip6t_LOG.c
index e6af8d7..683550e 100644
--- a/net/ipv6/netfilter/ip6t_LOG.c
+++ b/net/ipv6/netfilter/ip6t_LOG.c
@@ -368,7 +368,7 @@ static void dump_packet(struct sbuff *m,
 	}
 
 	/* Max length: 16 "MARK=0xFFFFFFFF " */
-	if (!recurse && skb->mark)
+	if (recurse && skb->mark)
 		sb_add(m, "MARK=0x%x ", skb->mark);
 }
 
-- 
1.8.1.4

